app/src/main/java/org/fossasia/badgemagic/ui/fragments/TextArtFragment.kt

```diff
@@ -378,7 +379,11 @@ class TextArtFragment : BaseFragment() {
     private fun putBitmapInEditText(drawableInfo: DrawableInfo) {
         val strToAppend = "«${drawableInfo.id}»"
         val spanStringBuilder = SpannableStringBuilder(strToAppend)
-        spanStringBuilder.setSpan(CenteredImageSpan(requireContext(), ImageUtils.trim(ImageUtils.vectorToBitmap(drawableInfo.image as VectorDrawable), 70)), 0, strToAppend.length, 33)
+        spanStringBuilder.setSpan(CenteredImageSpan(requireContext(), ImageUtils.trim(
+            if (drawableInfo.image is VectorDrawable)
+                ImageUtils.vectorToBitmap(drawableInfo.image)
+            else
+                (drawableInfo.image as BitmapDrawable).bitmap, 70)), 0, strToAppend.length, 33)
```

Comment:
use smart cast instead: `drawableInfo.image is BitmapDrawable`

Reply:
There are 3 more unsafe casts like that.

Reply:
> There are 3 more unsafe casts like that.

on it

Reply:
done, please review